Common/include/option_structure.hpp

```diff
@@ -466,6 +467,8 @@ const int TRANS_SOL = 4;	/*!< \brief Position of the transition model solution i
 const int HEAT_SOL = 5;		/*!< \brief Position of the heat equation in the solution solver array. */
 const int ADJHEAT_SOL = 6;  /*!< \brief Position of the adjoint heat equation in the solution solver array. */
 
+const int SCALAR_SOL = 5;    /*!< \brief Position of the scalar transport solution in the solver container array. */
```

Comment:
I think that, in the end, it would be great to make the current heat solver a solver for heat conduction in solids alone by removing the weakly coupled option. This will simplify things throughout the code a bit and let us remove some things. After/with this PR, it should be easier to have the heat solver inherit from some of the base scalar classes, as I have done for the turbulence models.